dotnet/src/IntegrationTests/Planning/StepwisePlanner/StepwisePlannerTests.cs

```diff
@@ -96,6 +98,49 @@ public async void CanExecuteStepwisePlan(bool useChatModel, string prompt, strin
         Assert.True(stepsTaken.Count >= expectedMinSteps && stepsTaken.Count <= 10, $"Actual: {stepsTaken.Count}. Expected at least {expectedMinSteps} steps and at most 10 steps to be taken.");
     }
 
+    [Fact]
+    public async void ExecutePlanFailsWithTooManyFunctions()
+    {
+        // Arrange
+        IKernel kernel = this.InitializeKernel();
+        var bingConnector = new BingConnector(this._bingApiKey);
+        var webSearchEngineSkill = new WebSearchEngineSkill(bingConnector);
+        kernel.ImportSkill(webSearchEngineSkill, "WebSearch");
+        kernel.ImportSkill(new TextSkill(), "text");
+        kernel.ImportSkill(new ConversationSummarySkill(kernel), "ConversationSummary");
+        kernel.ImportSkill(new MathSkill(), "Math");
+        kernel.ImportSkill(new FileIOSkill(), "FileIO");
+        kernel.ImportSkill(new HttpSkill(), "Http");
+
+        var planner = new Microsoft.SemanticKernel.Planning.StepwisePlanner(kernel, new() { MaxTokens = 1000 });
+
+        // Act
+        var plan = planner.CreatePlan("I need to buy a new brush for my cat. Can you show me options?");
+
+        // Assert
+        var ex = await Assert.ThrowsAsync<SKException>(async () => await plan.InvokeAsync().ConfigureAwait(false)).ConfigureAwait(false);
+        Assert.Equal("ChatHistory is too long to get a completion. Try reducing the available functions.", ex.Message);
+    }
+
+    [Fact]
+    public async void ExecutePlanSucceedsWithAlmostTooManyFunctions()
+    {
+        // Arrange
+        IKernel kernel = this.InitializeKernel();
+
+        _ = await kernel.ImportAIPluginAsync("Klarna", new Uri("https://www.klarna.com/.well-known/ai-plugin.json"), new OpenApiSkillExecutionParameters(enableDynamicOperationPayload: true)).ConfigureAwait(false);
```

Comment:
Will change this to something with a bit more control -- this was just the available repro.